applications/GeoMechanicsApplication/tests/cpp_tests/custom_constitutive/test_linear_elastic_plane_strain_2D_law.cpp

```diff
@@ -0,0 +1,201 @@
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "boost/numeric/ublas/assignment.hpp"
+#include "custom_constitutive/linear_elastic_plane_strain_2D_law.h"
+#include "custom_constitutive/plane_strain_type.h"
+#include "tests/cpp_tests/geo_mechanics_fast_suite.h"
+
+namespace
+{
+
+using namespace Kratos;
+
+Vector CalculateStress(GeoLinearElasticPlaneStrain2DLaw& rConstitutiveLaw)
+{
+    ConstitutiveLaw::Parameters parameters;
+    parameters.Set(ConstitutiveLaw::COMPUTE_CONSTITUTIVE_TENSOR);
+    parameters.Set(ConstitutiveLaw::COMPUTE_STRESS);
+    parameters.Set(ConstitutiveLaw::USE_ELEMENT_PROVIDED_STRAIN);
+
+    auto strain = Vector{ScalarVector{4, 1.0}};
+    parameters.SetStrainVector(strain);
+
+    Vector stress;
+    parameters.SetStressVector(stress);
+
+    Matrix constitutive_matrix;
+    parameters.SetConstitutiveMatrix(constitutive_matrix);
+
+    Properties properties;
+    properties.SetValue(YOUNG_MODULUS, 1.0e7);
+    properties.SetValue(POISSON_RATIO, 0.3);
+    parameters.SetMaterialProperties(properties);
+
+    rConstitutiveLaw.CalculateMaterialResponsePK2(parameters);
+
+    return stress;
+}
+
+GeoLinearElasticPlaneStrain2DLaw CreateLinearElasticPlaneStrainLaw()
+{
+    return GeoLinearElasticPlaneStrain2DLaw{std::make_unique<PlaneStrainType>()};
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawClone, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto       law   = CreateLinearElasticPlaneStrainLaw();
+    const auto clone = law.Clone();
+    KRATOS_EXPECT_NE(&law, clone.get());
+    KRATOS_EXPECT_NE(dynamic_cast<const GeoLinearElasticPlaneStrain2DLaw*>(clone.get()), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawRequiresInitializeResponse, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_TRUE(law.RequiresInitializeMaterialResponse())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawRequiresFinalizeResponse, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_TRUE(law.RequiresFinalizeMaterialResponse())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawIsIncremental, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+    KRATOS_EXPECT_TRUE(law.IsIncremental())
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GeoLinearElasticPlaneStrain2DLawReturnsExpectedLawFeatures, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    auto law = CreateLinearElasticPlaneStrainLaw();
+
+    ConstitutiveLaw::Features law_features;
+    law.GetLawFeatures(law_features);
+
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::PLANE_STRAIN_LAW))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::INFINITESIMAL_STRAINS))
+    KRATOS_EXPECT_TRUE(law_features.mOptions.IsDefined(ConstitutiveLaw::ISOTROPIC))
```

Comment:
Wouldn't we also want to check whether the options have the expected values? Now we only check whether the options have been defined, but not their value. For instance:
```cpp
    KRATOS_EXPECT_TRUE(law_features.mOptions.Is(ConstitutiveLaw::PLANE_STRAIN_LAW))
```

Reply:
Changed from IsDefined to Is